tests/components/vicare/test_config_flow.py

```diff
@@ -206,3 +206,22 @@ async def test_user_input_single_instance_allowed(hass: HomeAssistant) -> None:
     )
     assert result["type"] is FlowResultType.ABORT
     assert result["reason"] == "single_instance_allowed"
+
+
+async def test_options_flow(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test config flow options."""
+
+    await setup_integration(hass, mock_config_entry)
+    assert mock_config_entry.options.get(CONF_HEATING_TYPE) == HeatingType.auto.value
+
+    result = await hass.config_entries.options.async_init(mock_config_entry.entry_id)
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "init"
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"], user_input={CONF_HEATING_TYPE: HeatingType.gas.value}
+    )
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert mock_config_entry.options.get(CONF_HEATING_TYPE) == HeatingType.gas.value
```

Comment:
Maybe follow this test all the way so the integration reloads and perhaps we can see from some entity it's now gas instead of auto?

Reply:
Good idea, though would not know how to do that. Can I access the entities created for the config entry?

Reply:
As I have no idea which entity can reflect this and how below code is of course fake so you need to find the right entity and it's state for saying it's successful.
```diff
diff --git a/tests/components/vicare/test_config_flow.py b/tests/components/vicare/test_config_flow.py
index 3b0ffec8cd..067c7e7d51 100644
--- a/tests/components/vicare/test_config_flow.py
+++ b/tests/components/vicare/test_config_flow.py
@@ -233,5 +233,10 @@ async def test_options_flow(
     result = await hass.config_entries.options.async_configure(
         result["flow_id"], user_input={CONF_HEATING_TYPE: HeatingType.auto.value}
     )
+    await hass.async_block_till_done()
     assert result["type"] is FlowResultType.CREATE_ENTRY
     assert mock_config_entry.options.get(CONF_HEATING_TYPE) == HeatingType.auto.value
+
+    state = hass.states.get("sensor.xyz")
+    assert state
+    assert state.state == "something"
```

Reply:
I thought the integration would already be setup but maybe that's not entirely the case then (the entities should already be there). 
I'll take a new look tomorrow 👍

Reply:
Thanks, I copied over a test case from `test_climate.py` that should work, but it's failing if I insert it in that file.

Reply:
@gjohansson-ST any idea?

Reply:
I took the liberty for fixing it, see the last commit https://github.com/home-assistant/core/pull/128840/commits/f1398778da5d31af55b3091cd0d3b52fc12071d7